tv_apps/app-deck/js/promotion_list.js

```diff
@@ -0,0 +1,199 @@
+    this.animator = document.getElementById('app-deck-animation-banner');
+
+    this.animator.addEventListener('transitionend', this);
+
+    this._initBannerSize(this.animator);
+    this._initBannerSize(this.container);
+
+    this.animator.classList.add('hidden');
+
+    // load test content.
+    TEST_LIST.forEach(this._createBanner.bind(this));
+    // set first one as context item.
+    this.setContextItem(0);
+  };
+
+  proto._createBanner = function(item, index) {
+    var div = document.createElement('div');
+    div.classList.add('app-banner');
+
+    var btn = document.createElement('smart-button');
+    btn.setAttribute('type', 'banner-button');
+    btn.classList.add('app-deck-list-item');
+    btn.style.backgroundImage = 'url(\'' + item.imageUrl + '\'), ' +
+                                'url(\'style/icons/ic_default_app.png\')';
+    btn.dataset.index = index;
+
+    btn.addEventListener('focus', this);
+
+    div.appendChild(btn);
+    this.bannerContainer.appendChild(div);
+  };
+
+  proto._initBannerSize = function pl_initBannerSize(container) {
+    var bg = container.querySelector('.banner-background');
+    var text = container.querySelector('.banner-text-container');
+
+    text.style.width = bg.style.width = container.clientWidth + 'px';
+  };
+
+  proto._fillMainBanner = function pl_fillMainBanner(container, item) {
+    var bg = container.querySelector('.banner-background');
+    var title = container.querySelector('.banner-title');
+    var desc = container.querySelector('.banner-description');
+    bg.style.backgroundImage = 'url(\'' + item.imageUrl + '\')';
+    title.textContent = item.title;
+    desc.textContent = item.description;
+  };
+
+  proto._slideItemIn = function pl_slideItemIn(item) {
+    this.animator.classList.remove('hidden');
+    this.animator.classList.add('standby');
+
+    // fill animation and start to slide
+    this._fillMainBanner(this.animator, item);
+    // cancel previous one if we already have it.
+    this._cancelSliding();
+
+    this._animationTimeout = setTimeout(function() {
+      this._contextItem = item;
+      this.animationTimeout = 0;
+      this.animator.classList.remove('standby');
+      this.animator.classList.add('sliding');
+      this.container.classList.add('sliding');
+    }.bind(this), ANIMATION_TIMER);
+  };
+
+  proto._cancelSliding = function pl_cancelSliding() {
+    if (this._animationTimeout) {
+      clearTimeout(this._animationTimeout);
+      this._animationTimeout = 0;
+    }
+  };
+
+  proto.setContextItem = function pl_setContextItem(idx) {
+    var item = TEST_LIST[idx];
+
+    if (!this._contextItem) {
+      // If context item is undefined or null, it's first context item, we just
+      // set it without animation.
+      this._contextItem = item;
+      this._fillMainBanner(this.container, item);
+      return;
+    }
+
+    if (this._contextItem === item) {
+      this._cancelSliding();
+      console.log('the same item found: ' + idx);
```

Comment:
nit: I think we could remove console log of debugging purpose. what do you think?